frame/tips/src/lib.rs

```diff
@@ -379,6 +383,31 @@ decl_module! {
 			Tips::<T>::insert(&hash, tip);
 		}
 
+		/// Remove and slash an already-open tip.
+		///
+		/// May only be called from `T::RejectOrigin`.
+		///
+		/// As a result, API will slash the finder and the deposits are lost.
+		///
+		/// Emits `TipSlashed` if successful.
+		#[weight = 10_000]
+		fn slash_tip(origin, hash: T::Hash) {
+			T::RejectOrigin::ensure_origin(origin)?;
+
+			let tip = Tips::<T>::get(hash).ok_or(Error::<T>::UnknownTip)?;
```

Comment:
```suggestion
			let tip = Tips::<T>::take(hash).ok_or(Error::<T>::UnknownTip)?;
```
And no need to remove it later